.circleci/config.yml

```diff
@@ -0,0 +1,279 @@
+version: 2.1
+
+templates:
+  tagged-filter: &tagged-filter
+    tags:
+      only: /[0-9]+(\.[0-9]+)*/
+
+executors:
+  ubuntu-builder:
+    docker:
+      - image: trustlines/builder:master9
+        environment:
+          - SOLC_VERSION=v0.4.25
+    working_directory: ~/repo
+
+# define some common commands
+# see https://circleci.com/docs/2.0/configuration-reference/#commands-requires-version-21
+commands:
+  create-venv:
+    description: "Create venv"
+    steps:
+      - run:
+          name: Create python venv
+          command: |
+            python3 -m venv venv
+            pip install -U pip wheel setuptools
+
+  config-path:
+    description: "set environment variables and change PATH"
+    steps:
+    - run:
+        name: Configuring PATH
+        command: |
+          echo >> ${BASH_ENV} 'export PATH=~/bin:~/repo/venv/bin:${PATH}'
+          echo >> ${BASH_ENV} '. ~/.nvm/nvm.sh'
+jobs:
+  run-flake8:
+    executor: ubuntu-builder
+
+    steps:
+      - checkout
+      - config-path
+      - restore_cache:
+          key: v1-flake8-venv-{{ checksum "constraints.txt" }}
+      - create-venv
+      - run:
+          name: Install flake8
+          command: |
+            pip install flake8 flake8-string-format flake8-per-file-ignores -c constraints.txt
```

Comment:
I'm worried that this might get out of sync with the content of https://github.com/trustlines-network/contracts/blob/develop/requirements.txt and lead to differences between the local checks and the CI checks. For example I think we do not use those plugins locally at the moment.
We should use one source of truth here.